app/api/files/routes.ts

```
@@ -65,6 +67,54 @@ const filterByEntityPermissions = async (fileList: FileType[]): Promise<FileType
   return fileList.filter(f => !f.entity || allowedSharedIds.has(f.entity));
 };
 
+const isFilePubliclyAccessible = async (
+  file: FileType,
+  isPrivateInstance: boolean
+): Promise<boolean> => {
+  if (isPrivateInstance) {
+    return false;
+  }
+
+  if (file.type === 'custom') {
+    return true;
+  }
+
+  if (!file.entity) {
+    return false;
+  }
+
+  try {
+    const relatedEntities: EntitySchema[] = await entities.get(
+      { sharedId: file.entity },
+      'published',
+      { withoutDocuments: true }
+    );
+
```

I am not sure about this try/catch block. Always weary of a catch block with silent behavior (although it returns false). What would be the logic and reason for not passing the try block?

---

We should not leave it like this, an error is an error, and transforming that error in a response like `false` is going to be a potential pain, let the error bubble up. 
In case there are specific known operational errors then conditionally catch them on the catch block.